pkg/services/ngalert/migration/service.go

```diff
@@ -6,24 +6,43 @@ import (
 	"fmt"
 	"time"
 
+	alertingModels "github.com/grafana/alerting/models"
+	v2 "github.com/prometheus/alertmanager/api/v2"
+	"github.com/prometheus/common/model"
+	"golang.org/x/sync/semaphore"
+
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/infra/serverlock"
+	legacymodels "github.com/grafana/grafana/pkg/services/alerting/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	migmodels "github.com/grafana/grafana/pkg/services/ngalert/migration/models"
 	migrationStore "github.com/grafana/grafana/pkg/services/ngalert/migration/store"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/secrets"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 // actionName is the unique row-level lock name for serverlock.ServerLockService.
 const actionName = "alerting migration"
 
+var ErrUpgradeInProgress = errors.New("upgrade in progress")
+
 type UpgradeService interface {
 	Run(ctx context.Context) error
+	MigrateAlert(ctx context.Context, orgID int64, dashboardID int64, panelID int64) (definitions.OrgMigrationSummary, error)
+	MigrateDashboardAlerts(ctx context.Context, orgID int64, dashboardID int64, skipExisting bool) (definitions.OrgMigrationSummary, error)
+	MigrateAllDashboardAlerts(ctx context.Context, orgID int64, skipExisting bool) (definitions.OrgMigrationSummary, error)
+	MigrateChannel(ctx context.Context, orgID int64, channelID int64) (definitions.OrgMigrationSummary, error)
+	MigrateAllChannels(ctx context.Context, orgID int64, skipExisting bool) (definitions.OrgMigrationSummary, error)
+	MigrateOrg(ctx context.Context, orgID int64, skipExisting bool) (definitions.OrgMigrationSummary, error)
+	GetOrgMigrationState(ctx context.Context, orgID int64) (*definitions.OrgMigrationState, error)
+	RevertOrg(ctx context.Context, orgID int64) error
```

Comment:
I think these methods make sense to declare in `api` package for the api handlers. It would save you from declaring a bunch of shallow methods for the fake migrator

Reply:
The purpose of the `fakeMigrationService` is so that I don't have to provide the real thing to the external tests `pkg/services/quota/quotaimpl/quota_test.go` and `pkg/services/ngalert/tests/util.go`. So, I can move the interface to the api package but it won't save me from declaring the shallow methods on the fake.